### PR TITLE
Feat/movj action server 

### PR DESCRIPTION
**Pull request by Mt-is-at-O**
# 変更点
MovJをserviceではなくactionで呼び出せるようにActionServerを実装した。
`ros2 action send_goal /mg400/mov_j mg400_msgs/action/MovJ {}`
より呼び出し可能
RobotModeがENABLEの状態、目標の位置に到達、この２つを満たす時actionの動作が終了となる。

### CODE REVIEW

**Comment by Mt-is-at-O** [comment]
１つ目のターミナル
`ros2 launch mg400_bringup mg400.launch.py`
２つ目のターミナル
`. sample.bash`
よりactionを用いた動きを見れる

**Comment by m12watanabe1a** [comment]
@Mt-is-at-O
Thank you for the implementation.
Your code was almost perfect. And I changed a bit for implementation simplicity.


And note that the`mov_j` msg interface has been changed by this PR,
So the other repos will be affected.

File: mg400_bringup/launch/mg400.launch.py [hunk 1]
```diff
@@ -61,6 +66,15 @@ def generate_launch_description():
...
             {'ip_address': ip_address,
              'service_level': service_level}])
+    action_node = Node(
+        package='mg400_node',
+        executable='action_node_exec',
+        namespace=namespace,
+        name='mg400_action_node',
+        on_exit=Shutdown(),
+        parameters=[
+            {'ip_address': ip_address,
```
Comment (m12watanabe1a) [inline mg400_bringup/launch/mg400.launch.py:76]: This is not needed

File: mg400_bringup/launch/mg400.launch.py [hunk 2]
```diff
@@ -61,6 +66,15 @@ def generate_launch_description():
...
+             'action_level': action_level}])
```
Comment (m12watanabe1a) [inline mg400_bringup/launch/mg400.launch.py:77]: This also is not needed for now

File: mg400_bringup/launch/mg400.launch.py [hunk 3]
```diff
@@ -87,8 +101,10 @@ def generate_launch_description():
     ld.add_action(ip_address_arg)
     ld.add_action(joy_arg)
     ld.add_action(service_level_arg)
+    ld.add_action(action_level_arg)
```
Comment (m12watanabe1a) [inline mg400_bringup/launch/mg400.launch.py:104]: this also

File: mg400_bringup/launch/simulator.launch.py [hunk 4]
```diff
@@ -42,12 +47,14 @@ def generate_launch_description():
         launch_arguments=[
             ('ip_address', '127.0.0.1'),
             ('joy', joy),
-            ('service_level', service_level)])
+            ('service_level', service_level),
+            ('action_level', action_level)])
```
Comment (m12watanabe1a) [inline mg400_bringup/launch/simulator.launch.py:51]: This also

**Review by m12watanabe1a — CHANGES_REQUESTED** [review]

File: mg400_node/src/action_node.cpp [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+  const double tolerance)
+{
+  auto is_in_tolerance = [tolerance](
+    const double val) -> bool {
+      return std::abs(val) < tolerance;
+    };
+
+  return is_in_tolerance(this->current_end_pose_.x - goal.x) &&
+         is_in_tolerance(this->current_end_pose_.y - goal.y) &&
+         is_in_tolerance(this->current_end_pose_.z - goal.z);
```
Comment (m12watanabe1a) [inline mg400_node/src/action_node.cpp:165]: `current_end_pose.r` also should be checked in future PR

**Review by m12watanabe1a — APPROVED** [review]

**Comment by m12watanabe1a** [comment]
We need check this PR with real HW again

**Comment by m12watanabe1a** [comment]
Now it is confirmed to work

File: mg400_node/src/action_node.cpp (hunk 5 above)
Comment (m12watanabe1a) [reply mg400_node/src/action_node.cpp]: This requirements has been solved in this PR.

**Pull request merged**